Sources/MapboxCoreNavigation/HistoryRecorder.swift

```diff
@@ -5,13 +5,15 @@ class HistoryRecorder {
     /**
      Path to the directory where history file could be stored when `HistoryRecording.stopRecordingHistory(writingFileWith:)` is called.
      
-     Setting `nil` disables history recording. Defaults to `nil`.
+     Setting `nil` disables history recording. Defaults to `nil`. Updating value from `nil` to `non-nil` value results in recreating the shared instance since `nil` guaranteed an invalid handler. Further updates have no effect.
```

Comment:
Didn't want to add it because it's a public protocol user may potentially implement himself (for whatever reason) and thus he may introduce his own logic around directory configuring. But I guess it is unlikely to happen and then user will be aware of his own decisions...